### PR TITLE
Fix returning the namespace for listing resources

### LINKED ISSUE #86: listing services and revisions should respect namespace
```
➜  kubectl create namespace n2        
namespace/n2 created

➜./kn service create svc1 -n n2 --image dev.local/ns/i:v1    

# bug: list without any option returning services from all the namespaces
➜./kn service list
s1
s2
svc1

➜ ./kn service list -n n2
svc1

➜ ./kn service list -n default
s1
s2
```

### PR DESCRIPTION
**Pull request by navidshaikh**
Fixes #86

### CODE REVIEW

File: pkg/kn/commands/namespaced.go [hunk 1]
```diff
@@ -31,19 +31,13 @@ func AddNamespaceFlags(flags *pflag.FlagSet, allowAll bool) {
...
 func GetNamespace(cmd *cobra.Command) (string, error) {
 	namespace := cmd.Flag("namespace").Value.String()
-	if cmd.Flags().Lookup("all-namespaces") == nil {
-		if namespace == "" {
-			return defaultNamespace, nil
-		}
-		return namespace, nil
-	}
-
 	all, err := cmd.Flags().GetBool("all-namespaces")
```
Comment (navidshaikh) [inline pkg/kn/commands/namespaced.go]: Well, this wont work for `service create` as it doesn't define the `all-namespaces` flag. My bad, updating the PR.

Comment (navidshaikh) [reply pkg/kn/commands/namespaced.go]: fixed

**Comment by cppforlife** [comment]
@navidshaikh should we add a unit test for this?

**Pull request merged**